phase-4-explainer.md

```diff
@@ -0,0 +1,175 @@
+```
+
+ARIA reflects these states by allowing an author to set the checked state with `aria-checked`, assigning it to a string value of either `true`, `false`, or `mixed` (the latter reflecting the indeterminate state). This mixed checked state can be used to indicate that a parent checkbox has some nested child checkboxes that are a combination of `checked` and `unchecked`. What should a `ComputedAccessibleNode` expose these three visual states? However, as a checkbox cannot be submitted with a value of `mixed` or `indeterminate` should this be stored in a separate attribute? Three possible options are:
+1. Mimic `aria-checked`and return a string indicating which state the element is in.
+2. Expose an enum for each state.
+3. Implement the `checked` attribute as a boolean, with a separate boolean attribute for `mixed`/`indeterminate`.
+
+One thing that needs to be agreed upon is the full set of attributes that a `ComputedAccessibleNode` should provide read access to. Is this the same as the set of properties available to be set via the `AccessibleNode` or `Element` interface? The current proposed IDL is:
+
+```
+interface ComputedAccessibleNode {
+    readonly attribute DOMString? autocomplete;
+    readonly attribute DOMString? checked;
+    readonly attribute DOMString? keyShortcuts;
+    readonly attribute DOMString? name;
+    readonly attribute DOMString? placeholder;
+    readonly attribute DOMString? role;
+    readonly attribute DOMString? roleDescription;
+    readonly attribute DOMString? valueText;
+
+    readonly attribute long? colCount;
+    readonly attribute unsigned long? colIndex;
+    readonly attribute unsigned long? colSpan;
+    readonly attribute unsigned long? level;
+    readonly attribute unsigned long? posInSet;
+    readonly attribute long? rowCount;
+    readonly attribute unsigned long? rowIndex;
+    readonly attribute unsigned long? rowSpan;
+    readonly attribute long? setSize;
+
+    readonly attribute ComputedAccessibleNode? parent;
+    readonly attribute ComputedAccessibleNode? firstChild;
+    readonly attribute ComputedAccessibleNode? lastChild;
+    readonly attribute ComputedAccessibleNode? previousSibling;
+    readonly attribute ComputedAccessibleNode? nextSibling;
+
+    readonly attribute boolean? atomic;
+    readonly attribute boolean? busy;
+    readonly attribute boolean? disabled;
+    readonly attribute boolean? modal;
+    readonly attribute boolean? readOnly;
+
+  readonly attribute float? valueNow;
+  readonly attribute float? valueMin;
+  readonly attribute float? valueMax;
+
+    [CallWith=ScriptState] Promise ensureUpToDate();
+};
+```
+
+A full list of available attributes can be found [on this trix]() (aboxhall@: should this list be migrated somewhere it can be viewed externally? Putting it into the explainer seems wrong, should it be uploaded as a separate markdown or filed into an issue on the wicg page once the explainer is updated there?).
```

Comment:
We should make a draft spec.

Reply:
I can get started on that this week.

Reply:
Yeah! I would copy the original phase 1 spec and modify as needed.

Reply:
Okay, I'll add that in a new branch.